include/boost/geometry/views/reversible_view.hpp

```diff
@@ -22,20 +22,57 @@
 #include <boost/version.hpp>
 #include <boost/range/adaptor/reversed.hpp>
 
+#include <boost/geometry/core/point_order.hpp>
 #include <boost/geometry/core/ring_type.hpp>
 #include <boost/geometry/core/tag.hpp>
 #include <boost/geometry/core/tags.hpp>
 
+#include <boost/geometry/util/order_as_direction.hpp>
+
 #include <boost/geometry/views/identity_view.hpp>
 
 namespace boost { namespace geometry
 {
 
-/*!
-\brief Flag for iterating a reversible_view in forward or reverse direction
-\ingroup views
-*/
-enum iterate_direction { iterate_forward, iterate_reverse };
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail
+{
+
+// Other possible names:
+//   conditionally_reverse_view
```

Comment:
I would keep `reversible_view`
Or maybe it's not the best name.
So we want to view a range it as clockwise, regardless of the underlying orientation.
`clockwise_view` as you have somewhere?

Reply:
I'd prefer `clockwise_view` too but this view takes `Direction` which either means "reverse" or "forward" and this has nothing to do with point order semantically. For this view to be replaced with `clockwise_view` we'd have to remove all instances of `Reverse` to `Direction` translation from the library and replace these with a view taking `Order` directly from Geometry.

Well, I guess we could replace `Reverse` to `Direction` with `Reverse` to `Order` but this would be confusing.